sympy/printing/ccode.py

```diff
@@ -219,9 +219,36 @@ def _print_ITE(self, expr):
         return self._print(_piecewise)
 
     def _print_MatrixElement(self, expr):
-        return "{0}[{1}]".format(expr.parent, expr.j +
+        from sympy.matrices import MatrixSymbol, MatrixExpr
+
+        if isinstance(expr.parent, MatrixSymbol):
+            return "{0}[{1}]".format(expr.parent, expr.j +
                 expr.i*expr.parent.shape[1])
 
+        terms = (expr.args[0].args)
+        PREC = precedence(expr.args[0])
+        l = []
+        for term in terms:
+            t = self._print(term)
+            if t.startswith('-'):
```

Comment:
I am just trying to keep track of the signs here. I plan to correct the `(-1)*B` bug in a separate PR.